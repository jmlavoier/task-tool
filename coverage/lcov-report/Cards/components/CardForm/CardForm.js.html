<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Cards/components/CardForm/CardForm.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">Cards/components/CardForm</a> CardForm.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">6.25% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>1/16</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">6.25% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>1/16</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
&nbsp;
import style from './CardForm.sass';
&nbsp;
class CardForm extends React.Component {
<span class="fstat-no" title="function not covered" >  co</span>nstructor(props) {
<span class="cstat-no" title="statement not covered" >    super(props);</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    this.state = {</span>
      description: this.props.description,
    };
&nbsp;
<span class="cstat-no" title="statement not covered" >    this.onDescriptionChange = this.onDescriptionChange.bind(this);</span>
<span class="cstat-no" title="statement not covered" >    this.onDescriptionKeyDown = this.onDescriptionKeyDown.bind(this);</span>
<span class="cstat-no" title="statement not covered" >    this.onDescriptionBlur = this.onDescriptionBlur.bind(this);</span>
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  co</span>mponentDidMount() {
<span class="cstat-no" title="statement not covered" >    if (this.el) {</span>
<span class="cstat-no" title="statement not covered" >      this.el.focus();</span>
    }
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  co</span>mponentDidUpdate() {
<span class="cstat-no" title="statement not covered" >    if (this.el) {</span>
<span class="cstat-no" title="statement not covered" >      this.el.focus();</span>
    }
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  on</span>DescriptionChange(event) {
<span class="cstat-no" title="statement not covered" >    this.setState({</span>
      description: event.target.value,
    });
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  on</span>DescriptionKeyDown(event) {
<span class="cstat-no" title="statement not covered" >    if (event.keyCode === 13) {</span>
<span class="cstat-no" title="statement not covered" >      this.props.onSaveCard({</span>
        id: this.props.id,
        description: this.state.description,
      });
    }
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  on</span>DescriptionBlur() {
<span class="cstat-no" title="statement not covered" >    this.props.onSaveCard({</span>
      id: this.props.id,
      description: this.state.description,
    });
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  re</span>nder() {
<span class="cstat-no" title="statement not covered" >    return (</span>
      &lt;textarea
        ref={<span class="fstat-no" title="function not covered" >(e</span>l) =&gt; { <span class="cstat-no" title="statement not covered" >this.el = el; </span>}}
        className={style['card-form']}
        value={this.state.description}
        onChange={this.onDescriptionChange}
        onKeyDown={this.onDescriptionKeyDown}
        onBlur={this.onDescriptionBlur}
      /&gt;
    );
  }
}
&nbsp;
CardForm.propTypes = {
  id: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  onSaveCard: PropTypes.func.isRequired,
};
&nbsp;
export default CardForm;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Nov 04 2017 19:30:25 GMT-0200 (-02)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
