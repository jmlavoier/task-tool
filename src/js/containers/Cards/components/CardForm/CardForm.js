import React from 'react';
import PropTypes from 'prop-types';

import style from './CardForm.sass';

class CardForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      description: this.props.description,
    }
  }

  onDescriptionChange(event) {
    this.setState({
      description: event.target.value,
    })
  }

  onDescriptionKeyDown(event) {
    if (event.keyCode === 13) {
      this.props.onSaveCard({
        id: this.props.id,
        description: this.state.description,
      });
    }
  }

  onDescriptionBlur() {
    this.props.onSaveCard({
      id: this.props.id,
      description: this.state.description,
    });
  }

  componentDidMount() {
    if (this.el) {
      this.el.focus();
    }
  }

  componentDidUpdate() {
    if (this.el) {
      this.el.focus();
    }
  }

  render() {
    return (
      <textarea 
        ref={el => this.el = el}
        className={style['card-form']}
        value={this.state.description} 
        onChange={this.onDescriptionChange.bind(this)} 
        onKeyDown={this.onDescriptionKeyDown.bind(this)}
        onBlur={this.onDescriptionBlur.bind(this)}
      />
    );
  }
}

export default CardForm;