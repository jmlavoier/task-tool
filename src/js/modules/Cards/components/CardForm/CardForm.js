import React from 'react';
import PropTypes from 'prop-types';

import style from './CardForm.sass';

class CardForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      description: this.props.description,
    };

    this.onDescriptionChange = this.onDescriptionChange.bind(this);
    this.onDescriptionKeyDown = this.onDescriptionKeyDown.bind(this);
    this.onDescriptionBlur = this.onDescriptionBlur.bind(this);
  }

  componentDidMount() {
    if (this.el) {
      this.el.focus();
    }
  }

  componentDidUpdate() {
    if (this.el) {
      this.el.focus();
    }
  }

  onDescriptionChange(event) {
    this.setState({
      description: event.target.value,
    });
  }

  onDescriptionKeyDown(event) {
    // if (event.keyCode === 13) {
    //   this.props.onSaveCard({
    //     id: this.props.id,
    //     description: this.state.description,
    //   });
    // }
  }

  onDescriptionBlur() {
    this.props.onSaveCard({
      id: this.props.id,
      description: this.state.description,
    });
  }

  render() {
    return (
      <textarea
        ref={(el) => { this.el = el; }}
        className={style['card-form']}
        value={this.state.description}
        onChange={this.onDescriptionChange}
        onKeyDown={this.onDescriptionKeyDown}
        onBlur={this.onDescriptionBlur}
      />
    );
  }
}

CardForm.propTypes = {
  id: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  onSaveCard: PropTypes.func.isRequired,
};

export default CardForm;
