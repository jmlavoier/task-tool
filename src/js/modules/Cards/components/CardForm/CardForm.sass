.card-form
  outline: none
  border: 0
  background-color: rgba(0,0,0,0.2)
  width: calc(100% - 5px)
  height: 100px
  resize: none
  color: #f1f1f1
  font-size: 14px
  border-radius: 5px