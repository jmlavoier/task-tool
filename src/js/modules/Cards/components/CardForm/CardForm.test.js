import React from 'react';
import { shallow, mount } from 'enzyme';

import CardForm from './CardForm';

test('Should component render with default props', () => {
  const id = '';
  const description = '';
  const wrapper = shallow(<CardForm 
      id={id}
      description={description}
      onSaveCard={() => {}}
  />);

  expect(wrapper).toMatchSnapshot();
});

test('Should component call onSaveCard on blur', () => {
  const id = '123';
  const description = 'Testing';
  const mockOnSaveCard = jest.fn();

  const wrapper = shallow(<CardForm 
      id={id}
      description={description}
      onSaveCard={mockOnSaveCard}
  />);

  wrapper.simulate('blur');
  expect(mockOnSaveCard).toHaveBeenCalled();
  expect(mockOnSaveCard.mock.calls[0][0]).toEqual({id, description });
});


test('Should component call onSaveCard on keydown Enter', () => {
  const id = '123';
  const description = 'Testing';
  const mockOnSaveCard = jest.fn();

  const wrapper = mount(<CardForm
    id={id}
    description={description}
    onSaveCard={mockOnSaveCard}
  />);

  wrapper.simulate('keydown', { keyCode: 13 });
  expect(mockOnSaveCard).toHaveBeenCalled();
  expect(mockOnSaveCard.mock.calls[0][0]).toEqual({id, description });
});


test('Should component doesn\'t call onSaveCard on keydown any key', () => {
  const id = '123';
  const description = 'Testing';
  const mockOnSaveCard = jest.fn();

  const wrapper = mount(<CardForm
    id={id}
    description={description}
    onSaveCard={mockOnSaveCard}
  />);

  wrapper.simulate('keydown', { keyCode: 40 });
  expect(mockOnSaveCard).not.toHaveBeenCalled();
});


test('Should component change description', () => {
  const id = '123';
  const description = 'Testing';
  const mockOnSaveCard = jest.fn();

  const wrapper = mount(<CardForm
    id={id}
    description={description}
    onSaveCard={mockOnSaveCard}
  />);
  expect(wrapper).toMatchSnapshot();
  
  wrapper.simulate('change', { target: { value: 'Description changed' } });
  expect(wrapper).toMatchSnapshot();
  
});