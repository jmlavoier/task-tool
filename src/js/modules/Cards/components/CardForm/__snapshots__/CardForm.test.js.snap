// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should component change description 1`] = `
<CardForm
  description="Testing"
  id="123"
  onSaveCard={[Function]}
>
  <textarea
    className="card-form"
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    value="Testing"
  />
</CardForm>
`;

exports[`Should component change description 2`] = `
<CardForm
  description="Testing"
  id="123"
  onSaveCard={[Function]}
>
  <textarea
    className="card-form"
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    value="Description changed"
  />
</CardForm>
`;

exports[`Should component render with default props 1`] = `
<textarea
  className="card-form"
  onBlur={[Function]}
  onChange={[Function]}
  onKeyDown={[Function]}
  value=""
/>
`;
